packages/@webex/plugin-meetings/test/unit/spec/meeting/muteState.js

```diff
@@ -509,3 +509,202 @@ describe('plugin-meetings', () => {
+          assert.isTrue(muteState.state.client.localMute);
+        });
+
+          
+        it('tests when track muted is false and remoteMuted is false', async () => {
+          setup(mediaType, false, false);
+          setupSpies(muteState, mediaType);
+          muteState.state.server.localMute = true;
+  
+          muteState.init(meeting);
+  
+          assert.calledWith(setUnmuteAllowedSpy, muteState.state.server.unmuteAllowed);
+          assert.notCalled(setServerMutedSpy);
+          assert.calledOnce(MeetingUtil.remoteUpdateAudioVideo);
+          assert.isFalse(muteState.state.client.localMute);
+        });
+  
+        it('tests when remoteMuted is true', async () => {
+          // testing that muteLocalTrack is called
+          setup(mediaType, true);
+          setupSpies(muteState, mediaType);
+  
+          muteState.init(meeting);
+  
+          const serverMutedSpyCall1 = setServerMutedSpy.getCall(0);
+          const serverMutedSpyCall2 = setServerMutedSpy.getCall(1);
+  
+          assert.calledWith(setUnmuteAllowedSpy, muteState.state.server.unmuteAllowed);
+          assert.equal(serverMutedSpyCall1.calledWithExactly(true, 'remotelyMuted'), true);
+          assert.equal(serverMutedSpyCall2.calledWithExactly(true, 'remotelyMuted'), true);
+        });
+      });
+
+      describe('#handleLocalTrackMuteStateChange', () => {
+
+        afterEach(() => {
+          sinon.restore();
+        });
+
+        it('checks when ignoreMuteStateChange is true nothing changes', () => {
+          setup(mediaType, false, false);
+          muteState.ignoreMuteStateChange= true;
+
+          muteState.handleLocalTrackMuteStateChange(meeting, true);
+          assert.notCalled(MeetingUtil.remoteUpdateAudioVideo);
+
+          assert.isFalse(muteState.state.client.localMute);
+        });
+
+        it('tests localMute - true to false', () => {
+          setup(mediaType, false, true);
+
+          muteState.handleLocalTrackMuteStateChange(meeting, false);
+          assert.equal(muteState.state.client.localMute, false);
+          assert.called(MeetingUtil.remoteUpdateAudioVideo);
+        });
+
+        it('tests localMute - false to true', () => {
+          setup(mediaType, false, false);
+
+          muteState.handleLocalTrackMuteStateChange(meeting, true);
+          assert.equal(muteState.state.client.localMute, true);
+          assert.called(MeetingUtil.remoteUpdateAudioVideo);
+        });
+      });
+
+      describe('#applyClientStateLocally', () => {
+
+        afterEach(() => {
+          sinon.restore();
+        });
+
+        it('checks when sdkOwnsLocalTrack is false', () => {
+          setup(mediaType);
+          setupSpies(muteState, mediaType);
+          muteState.sdkOwnsLocalTrack= false;
+
+          muteState.applyClientStateLocally(meeting, 'somereason');
+          assert.calledOnceWithExactly(setServerMutedSpy, muteState.state.client.localMute, 'somereason');
```

Comment:
it would be good to check that setMuted was not called here

Reply:
done